cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Polysemy.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Test.Cardano.CLI.Polysemy
+  ( cardanoCliPath
+  , execCardanoCli
+  , execCardanoCli_
+  , localWorkspace
+  ) where
+
+import           HaskellWorks.Polysemy
+import           HaskellWorks.Polysemy.Error.Types
+import           HaskellWorks.Polysemy.Hedgehog
+import           HaskellWorks.Polysemy.Hedgehog.Assert
+import           HaskellWorks.Polysemy.Hedgehog.Process
+import           HaskellWorks.Polysemy.Prelude
+
+cardanoCliPath :: FilePath
+cardanoCliPath = "cardano-cli"
+
+-- | Execute cardano-cli via the command line.
+--
+-- Waits for the process to finish and returns the stdout.
+execCardanoCli :: ()
+  => HasCallStack
+  => Member (Embed IO) r
+  => Member Hedgehog r
+  => Member Log r
+  => [String]
+  -- ^ Arguments to the CLI command
+  -> Sem r String
+  -- ^ Captured stdout
+execCardanoCli args = withFrozenCallStack $
+  execFlexOk "cardano-cli" "CARDANO_CLI" args
+    & trapFail @GenericError
+    & trapFail @IOException
+
+execCardanoCli_ :: ()
+  => HasCallStack
+  => Member (Embed IO) r
+  => Member Hedgehog r
+  => Member Log r
+  => [String]
+  -- ^ Arguments to the CLI command
+  -> Sem r ()
+execCardanoCli_ args = withFrozenCallStack $
+  void $ execCardanoCli args
+
+localWorkspace ::  ()
+  => Member Hedgehog r
+  => Member Log r
+  => Member (Embed IO) r
+  => Sem
+        ( Reader Workspace
+        : Reader ProjectRoot
+        : Reader PackagePath
+        : Resource
+        : r)
```

Comment:
Wouldn't it be more consistent to encode these are `Member` constraints? Or maybe it's not possible for some reason?